### PR TITLE
Add clarification about OpenSearch ingest pipelines compared to Data …

### LINKED ISSUE #6429: [DOC] Updates to ingestion/Data Prepper documentation
**What do you want to do?**
 Request a change to existing documentation

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Update the ingestion documentation to clarify Data Prepper concepts. 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
- [x] Update ['ingest pipelines' ](https://opensearch.org/docs/latest/ingest-pipelines/)intro to clarify the difference between these pipelines and Data Prepper. For example, something like "The OpenSearch ingest pipelines run on the OpenSearch cluster. We recommend you only use them for modifications you need to make on the cluster. For other last-mile changes, we recommend using Data Prepper which runs off the cluster." (#6626) and (#6786 )
- [ ] For the individual ingest pipelines pages, describe them as 'index pipelines' since they perform actions on indexes. See https://opensearch.org/docs/latest/ingest-pipelines/get-ingest/ **No change to naming because it will cause code problems. See @gaobinlong comments.**
- [x] From the individual ingest pipelines pages, point them to the DP documentation for further information. **Certain ingest processors aren't available in Data Prepper, including machine learning and vector embedding**
- [x] From the Data Prepper home page, acknowledge that DP is the recommended application for ingestion in OpenSearch. https://opensearch.org/docs/latest/data-prepper/ 
- [ ] Revisit the Tools section to discuss ingestion tools: https://opensearch.org/docs/latest/tools/

### PR DESCRIPTION
**Pull request by vagimeli**
…Prepper

### Description
Clarifies preferred data ingestion tools for OpenSearch

### Issues Resolved
Closes parts of #6429 

### Checklist
- [ X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@dlvenable @gaobinlong Thank you for providing context about ingest pipelines and Data Prepper. Please review this PR at your availability. I'll be out of office Mar 26-April 2, so please reach out to @hdhalter if this PR needs changes or reviews before I return. Thank you, Melissa

**Comment by hdhalter** [comment]
Thanks, Melissa! I will update the label to 'tech review' for sign off by @dlvenable and @gaobinlong.

**Comment by hdhalter** [comment]
Tagging @Naarcha-AWS for doc review.

**Comment by vagimeli** [comment]
> Tagging @Naarcha-AWS for doc review.

Hi @Naarcha-AWS @dlvenable Please review this PR as soon as your schedule permits. Thank you.

**Review by dlvenable — APPROVED** [review]
Thank you for adding this clarity @vagimeli !

File: _data-prepper/index.md [hunk 1]
```diff
@@ -14,9 +14,9 @@ redirect_from:
 
 # Data Prepper
 
-Data Prepper is a server-side data collector capable of filtering, enriching, transforming, normalizing, and aggregating data for downstream analytics and visualization.
+Data Prepper is a server-side data collector capable of filtering, enriching, transforming, normalizing, and aggregating data for downstream analytics and visualization. Data Prepper is the preferred data ingestion tool for OpenSearch. It is recommended for most data ingestion use cases in OpenSearch and for processing large, complex datasets.
```
Comment (natebower) [inline _data-prepper/index.md:17]:
```suggestion
Data Prepper is a server-side data collector capable of filtering, enriching, transforming, normalizing, and aggregating data for downstream analysis and visualization. Data Prepper is the preferred data ingestion tool for OpenSearch. It is recommended for most data ingestion use cases in OpenSearch and for processing large, complex datasets.
```

File: _data-prepper/index.md [hunk 2]
```diff
@@ -14,9 +14,9 @@ redirect_from:
...
 
-Data Prepper lets users build custom pipelines to improve the operational view of applications. Two common uses for Data Prepper are trace and log analytics. [Trace analytics]({{site.url}}{{site.baseurl}}/observability-plugin/trace/index/) can help you visualize the flow of events and identify performance problems, and [log analytics]({{site.url}}{{site.baseurl}}/observability-plugin/log-analytics/) can improve searching, analyzing and provide insights into your application.
+With Data Prepper you can build custom pipelines to improve the operational view of applications. Two common uses for Data Prepper are trace analytics and log analytics. [Trace analytics]({{site.url}}{{site.baseurl}}/observability-plugin/trace/index/) can help you visualize events flow and identify performance problems. [Log analytics]({{site.url}}{{site.baseurl}}/observability-plugin/log-analytics/) can help improve searching, analyzing and give you deeper insights into your application.
```
Comment (natebower) [inline _data-prepper/index.md:19]:
```suggestion
With Data Prepper you can build custom pipelines to improve the operational view of applications. Two common use cases for Data Prepper are trace analytics and log analytics. [Trace analytics]({{site.url}}{{site.baseurl}}/observability-plugin/trace/index/) can help you visualize event flows and identify performance problems. [Log analytics]({{site.url}}{{site.baseurl}}/observability-plugin/log-analytics/) can help improve search and analytics and give you deeper insights into your application.
```

Comment (natebower) [inline _data-prepper/index.md:19]: The last sentence needs some clarification. Improve search and analytics what or how?

File: _ingest-pipelines/index.md [hunk 3]
```diff
@@ -16,7 +16,15 @@ An _ingest pipeline_ is a sequence of _processors_ that are applied to documents
 
 Processors are customizable tasks that run in a sequential order as they appear in the request body. This order is important, as each processor depends on the output of the previous processor. The modified documents appear in your index after the processors are applied.
 
-Ingest pipelines can only be managed using [ingest API operations]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/index/).
+## OpenSearch ingest pipelines compared to Data Prepper
+
+OpenSeach ingest pipelines run within the OpenSearch cluster, whereas [Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/) is an external component that runs off the OpenSearch cluster.
```
Comment (natebower) [inline _ingest-pipelines/index.md:21]:
```suggestion
OpenSeach ingest pipelines run within the OpenSearch cluster, whereas [Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/) is an external component that runs on the OpenSearch cluster. 
```

File: _ingest-pipelines/index.md [hunk 4]
```diff
@@ -16,7 +16,15 @@ An _ingest pipeline_ is a sequence of _processors_ that are applied to documents
 
 Processors are customizable tasks that run in a sequential order as they appear in the request body. This order is important, as each processor depends on the output of the previous processor. The modified documents appear in your index after the processors are applied.
 
-Ingest pipelines can only be managed using [ingest API operations]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/index/).
+## OpenSearch ingest pipelines compared to Data Prepper
+
+OpenSeach ingest pipelines run within the OpenSearch cluster, whereas [Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/) is an external component that runs off the OpenSearch cluster. 
+
+OpenSearch ingest pipelines perform actions on indexes and are preferred for use cases involving pre-processing simple datasets, [machine learning processors]({{site.url}}{{site.baseurl}}/ingest-pipelines/processors/sparse-encoding/), and [vector embedding processors]({{site.url}}{{site.baseurl}}/ingest-pipelines/processors/text-image-embedding/). OpenSearch ingest pipelines are recommended for simple data pre-processing and small datasets.
```
Comment (natebower) [inline _ingest-pipelines/index.md:23]:
```suggestion
OpenSearch ingest pipelines perform actions on indexes and are preferred for use cases involving pre-processing simple datasets, [machine learning (ML) processors]({{site.url}}{{site.baseurl}}/ingest-pipelines/processors/sparse-encoding/), and [vector embedding processors]({{site.url}}{{site.baseurl}}/ingest-pipelines/processors/text-image-embedding/). OpenSearch ingest pipelines are recommended for simple data pre-processing and small datasets. 
```

File: _ingest-pipelines/index.md [hunk 5]
```diff
@@ -16,7 +16,15 @@ An _ingest pipeline_ is a sequence of _processors_ that are applied to documents
...
 
-Ingest pipelines can only be managed using [ingest API operations]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/index/).
+## OpenSearch ingest pipelines compared to Data Prepper
+
+OpenSeach ingest pipelines run within the OpenSearch cluster, whereas [Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/) is an external component that runs off the OpenSearch cluster. 
+
+OpenSearch ingest pipelines perform actions on indexes and are preferred for use cases involving pre-processing simple datasets, [machine learning processors]({{site.url}}{{site.baseurl}}/ingest-pipelines/processors/sparse-encoding/), and [vector embedding processors]({{site.url}}{{site.baseurl}}/ingest-pipelines/processors/text-image-embedding/). OpenSearch ingest pipelines are recommended for simple data pre-processing and small datasets. 
+
+Data Prepper is the preferred data ingestion tool for OpenSearch. Data Prepper is recommended for any data processing that it supports and for use cases involving the transferring and fetching of large datasets and complex data pre-processing. Refer to the [Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/) documentation to learn more information.
```
Comment (natebower) [inline _ingest-pipelines/index.md:25]:
```suggestion
Data Prepper is the preferred data ingestion tool for OpenSearch and is recommended for any data processing that it supports and for use cases involving the transfer and fetching of large datasets and complex data pre-processing. Refer to the [Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/) documentation for more information.      
```

Comment (natebower) [inline _ingest-pipelines/index.md:25]: First sentence: This reads as though a noun should follow "data processing". Data processing what that it supports?

File: _ingest-pipelines/index.md [hunk 6]
```diff
@@ -16,7 +16,15 @@ An _ingest pipeline_ is a sequence of _processors_ that are applied to documents
...
+
+OpenSeach ingest pipelines run within the OpenSearch cluster, whereas [Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/) is an external component that runs off the OpenSearch cluster. 
+
+OpenSearch ingest pipelines perform actions on indexes and are preferred for use cases involving pre-processing simple datasets, [machine learning processors]({{site.url}}{{site.baseurl}}/ingest-pipelines/processors/sparse-encoding/), and [vector embedding processors]({{site.url}}{{site.baseurl}}/ingest-pipelines/processors/text-image-embedding/). OpenSearch ingest pipelines are recommended for simple data pre-processing and small datasets. 
+
+Data Prepper is the preferred data ingestion tool for OpenSearch. Data Prepper is recommended for any data processing that it supports and for use cases involving the transferring and fetching of large datasets and complex data pre-processing. Refer to the [Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/) documentation to learn more information.      
+
+OpenSearch ingest pipelines can only be managed using [ingest API operations]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/index/).
```
Comment (natebower) [inline _ingest-pipelines/index.md:27]: In the link, should "Ingest" be capitalized (is it the name of the API)?

File: _ingest-pipelines/processors/grok.md [hunk 7]
```diff
@@ -10,6 +10,9 @@ nav_order: 140
 
 The `grok` processor is used to parse and structure unstructured data using pattern matching. You can use the `grok` processor to extract fields from log messages, web server access logs, application logs, and other log data that follows a consistent format.
 
+This documentation is about using the `grok` processor in OpenSearch ingest pipelines. Consider using the [Data Prepper `grok` processor]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/grok/), which runs off the OpenSearch cluster, if your use case involves large or complex datasets.
```
Comment (natebower) [inline _ingest-pipelines/processors/grok.md:13]:
```suggestion
This documentation describes using the `grok` processor in OpenSearch ingest pipelines. Consider using the [Data Prepper `grok` processor]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/processors/grok/), which runs on the OpenSearch cluster, if your use case involves large or complex datasets.
```

**Review by natebower — COMMENTED** [review]
@vagimeli Please see my comments and changes and let me know if you have any questions. Thanks!

File: _data-prepper/index.md (hunk 2 above)
Comment (vagimeli) [reply _data-prepper/index.md]: Does this read better? `Log analytics equips you with tools to enhance your search capabilities, conduct comprehensive analysis, and gain profound insights into your applications' performance and behavior.`

Comment (natebower) [reply _data-prepper/index.md]: Perfect!

File: _ingest-pipelines/index.md (hunk 5 above)
Comment (vagimeli) [reply _ingest-pipelines/index.md]: Is this a better rewrite? `Data Prepper is recommended for any data processing tasks it supports, particularly when dealing with large datasets and complex data pre-processing requirements. It streamlines the process of transferring and fetching large datasets, while providing robust capabilities for intricate data preparation and transformation operations.`

Comment (natebower) [reply _ingest-pipelines/index.md]: Yes, that works, but no comma in the second sentence 😄

**Pull request merged**